xapian-bindings/java/java.i

```diff
@@ -189,6 +189,64 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%typemap(javainterfaces) Xapian::MSetIterator "Iterable<MSetIterator.MatchDescriptor>";
+%typemap(javaimports) Xapian::MSetIterator "import java.util.Iterator;"
+%typemap(javacode) MSetIterator %{
+  public Iterator<MatchDescriptor> iterator() {
+      Iterator<MatchDescriptor> iterator = new Iterator <MatchDescriptor>() {
+          @Override
+          public boolean hasNext() {
+              return MSetIterator.this.hasNext();
+          }
+          @Override
+          public MatchDescriptor next() {
+              MSetIterator thisIterator = MSetIterator.this;
+              long rank = thisIterator.getRank();
+              long docId = thisIterator.getDocId();
+              double weight = thisIterator.getWeight();
+              int percent = thisIterator.getPercent();
+              Document doc = thisIterator.getDocument();
+              MatchDescriptor descriptor = new MatchDescriptor(docId, rank, percent, weight, doc);
+	      thisIterator.next();
+              return descriptor;            
```

Comment:
To align this part with the rest of SWIG generated java file, I intentionally added the spaces. I fixed the thisIterator.next() line. I originally used tabs but the output java file was unaligned, so I switched to spaces. The spacing looks odd here but in the generated java file it aligns neatly.